packages/core-mobile/app/new/routes/(signedIn)/(tabs)/portfolio/index.tsx

```diff
@@ -174,17 +200,42 @@ const PortfolioHomeScreen = (): JSX.Element => {
         <ActionButtons buttons={ACTION_BUTTONS} />
       </View>
     )
-  }
+  }, [
+    ACTION_BUTTONS,
+    animatedHeaderStyle,
+    formattedBalance,
+    isLoading,
+    activeAccount?.name,
+    selectedCurrency,
+    theme.colors.$surfacePrimary,
+    totalPriceChanged,
+    indicatorStatus,
+    formattedPercent,
+    balanceAccurate,
+    handleBalanceHeaderLayout
+  ])
+
+  const handleSelectSegment = useCallback(
+    (index: number): void => {
+      selectedSegmentIndex.value = index
 
-  const handleSelectSegment = (index: number): void => {
-    if (tabViewRef.current?.getCurrentIndex() !== index) {
-      tabViewRef.current?.setIndex(index)
-    }
-  }
+      InteractionManager.runAfterInteractions(() => {
+        if (tabViewRef.current?.getCurrentIndex() !== index) {
+          tabViewRef.current?.setIndex(index)
+        }
+      })
+    },
+    [selectedSegmentIndex]
+  )
 
-  const handleChangeTab = (index: number): void => {
-    selectedSegmentIndex.value = index
-  }
+  const handleTabChange: OnTabChange = useCallback(
+    data => {
+      if (selectedSegmentIndex.value === data.prevIndex) {
+        selectedSegmentIndex.value = data.index
+      }
+    },
+    [selectedSegmentIndex]
```

Comment:
@onghwan also adjusted this logic so that we only set selectedSegmentIndex when necessary